app/lib/actions/katello/host/register.rb

```diff
@@ -29,7 +29,6 @@ def plan(host, system, consumer_params, content_view_environment, activation_key
             host.save!
 
             action_subject host
-            connect_to_smart_proxy(host)
```

Comment:
I looked at that a bit, but the finalize phase seemed like a better fit. In https://github.com/Dynflow/dynflow/blob/master/README.md#how-it-works, the doc mentions that finalize phase is for recording results into the DB, which is what `connect_to_smart_proxy` is doing.